components/StyledInputAmount.js

```diff
@@ -45,14 +46,10 @@ const getError = (curVal, minAmount, required) => {
   return Boolean((required && isNil(curVal)) || (minAmount && curVal < minAmount));
 };
 
-const i18nCurrencyName = (intl, currency) => {
-  const currencyNames = new Intl.DisplayNames(intl.locale, { type: 'currency' });
-  return currencyNames.of(currency);
-};
-
 const generateCurrencyOptions = (intl, availableCurrencies) => {
+  const currencyDisplayNames = getIntlDisplayNames(intl.locale, 'currency');
```

Comment:
@SudharakaP Move that here as re-using the same `Intl.DisplayNames` instance is [up to 6x faster](https://jsben.ch/JpWjB) than re-declaring it for each entry.